### ISSUE TITLE
[OpenBMC] xCAT clear BMC Dump logs when doing firmware flash

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#4327]
By request of the OpenBMC team, asking xCAT to do a clear of the BMC Dump entries before doing a firmware update on the BMC side

One place to trigger this is clear dumps before the reboot of the BMC if there is a pending firmware to be activate.  If there is none, we should not clear the dump on normal bmc reboot. 

PNOR updates should not trigger this one.

**Comment by whowutwut** [comment]
@zet809 @xuweibj Can we bubble this one to the top and see how much effort it would be to clear the logs before reboot of the BMC to activate a FW... 

It looks like as the machines are more long running, there are dumps that fill up the BMC and no new ones can be created at this time.  When I go to capture dump, I have to clear all first and re-generate.

**Comment by hu-weihua** [comment]
relate to task #4352

**Comment by whowutwut** [comment]
I looked into this a little today... it makes the code logic a bit fuzzy...  

Essentially....  reboot does the following: 
1. RPOWER_BMCREBOOT_REQUEST
2. RPOWER_BMCREBOOT_RESPONSE 

What we want to do now.... 
1. RINV_FIRM_REQUEST
2. RINV_FIRM_RESPONSE 
In here, we have some code logic... 
   ```
   If (bmcreboot) {
       if ($purpose_value =~ /BMC/ and priority == 0 and not exists ($functional->{$sw_id}) ) { 
          # This means that for a BMC firmware, it's set to activate, but it's not currently running as functional 
          set clear_dump = yes # for this machine 
          # Maybe here we set
          #    a) next_status to RSPCONFIG_DUMPCLRA_REQUEST
          #    b) next_status to RSPCONFIG_DUMPCLRA_RESPONSE
      }
   }
   ```
3. Clear DUMP
4. RPOWER_BMCREBOOT_REQUEST
5.  RPOWER_BMCREBOOT_RESPONSE


@xuweibj Is this something that could be implemented fairly quickly, or do we think this would take a lot more time? 

To UT, we have to:
1. take at least 2 nodes , activate BMC on 1 of the 2...
2. check dump list on the 2 nodes 
3. run `rpower <node range> bmcreboot` across the range
4. AFter bmc comes back up, check dump list on the 2 nodes, the activated BMC flash should be cleared

**Comment by whowutwut** [comment]
Another option is when we streamline the rflash command, we would do this, but it would also require similar changes here... and the stream line would get this function for "Free" if implemented

**Comment by xuweibj** [comment]
@whowutwut I agree with your idea.  I think this does not need too much time.

**Comment by whowutwut** [comment]
Will try out this code starting tomorrow and see what effects it has.

**Comment by whowutwut** [comment]
@xuweibj I'm not sure this code is working correctly.  (FYI @bybai) 

When I was testing the rflash code unattended with yesterdays build, it seems like looked at `rspconfig <> dump` we have Dump logs that are from Nov... those should have been cleared out? 

```
[root@briggs01 ~]# rspconfig p9up dump -l  | grep -v _debug
mid05tor12cn18: Error: [500] Login to BMC failed: 500 Can't connect to 172.11.139.18:443 (Network is unreachable).
mid05tor12cn18: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn05: Error: [500] Login to BMC failed: 500 Can't connect to mid05tor12cn05-bmc:443 (No route to host).
mid05tor12cn05: Error: BMC did not respond. Validate BMC configuration and retry the command.
mid05tor12cn13: [1] Generated: 11/28/2017 21:52:50, Size: 73576
mid05tor12cn13: [2] Generated: 11/28/2017 21:53:04, Size: 72812
mid05tor12cn13: [3] Generated: 11/28/2017 21:53:17, Size: 72664
mid05tor12cn13: [4] Generated: 11/28/2017 21:53:22, Size: 72540
mid05tor12cn13: [5] Generated: 11/28/2017 21:53:29, Size: 72700
mid05tor12cn13: [6] Generated: 11/28/2017 21:53:29, Size: 72832
mid05tor12cn13: [7] Generated: 11/28/2017 21:53:37, Size: 72552
mid05tor12cn13: [8] Generated: 11/28/2017 22:04:59, Size: 11656
mid05tor12cn13: [9] Generated: 11/28/2017 22:05:02, Size: 11692
mid05tor12cn13: [10] Generated: 11/30/2017 13:43:21, Size: 106920
mid05tor12cn13: [11] Generated: 11/30/2017 13:51:52, Size: 107328
mid05tor12cn13: [12] Generated: 11/30/2017 16:09:22, Size: 107684
mid05tor12cn13: [13] Generated: 11/30/2017 20:00:55, Size: 107876
mid05tor12cn13: [14] Generated: 11/30/2017 21:13:26, Size: 7860
mid05tor12cn13: [15] Generated: 12/01/2017 02:31:42, Size: 6888
mid05tor12cn13: [16] Generated: 12/01/2017 02:38:39, Size: 7004
mid05tor12cn13: [17] Generated: 12/01/2017 02:43:48, Size: 6856
mid05tor12cn13: [18] Generated: 12/01/2017 02:44:33, Size: 6812
mid05tor12cn13: [19] Generated: 12/01/2017 02:45:20, Size: 6784
mid05tor12cn13: [20] Generated: 12/01/2017 02:48:32, Size: 6780
mid05tor12cn13: [21] Generated: 12/01/2017 02:51:34, Size: 6740
mid05tor12cn13: [22] Generated: 12/01/2017 02:58:23, Size: 6788
mid05tor12cn13: [23] Generated: 12/01/2017 03:01:35, Size: 6772
mid05tor12cn16: [85] Generated: 12/01/2017 16:21:07, Size: 81572
mid05tor12cn16: [86] Generated: 12/01/2017 16:22:54, Size: 81868
mid05tor12cn16: [87] Generated: 12/01/2017 16:24:32, Size: 82088
mid05tor12cn16: [88] Generated: 12/01/2017 16:30:10, Size: 82324
mid05tor12cn16: [89] Generated: 12/01/2017 16:42:45, Size: 82528
mid05tor12cn16: [90] Generated: 12/01/2017 16:45:27, Size: 82816
mid05tor12cn16: [91] Generated: 12/03/2017 18:45:37, Size: 5144
mid05tor12cn16: [92] Generated: 12/04/2017 11:22:34, Size: 88296
mid05tor12cn16: [93] Generated: 12/04/2017 11:33:28, Size: 88496
mid05tor12cn16: [94] Generated: 12/04/2017 11:36:24, Size: 88800
mid05tor12cn16: [95] Generated: 12/04/2017 11:40:13, Size: 89048
mid05tor12cn16: [96] Generated: 12/05/2017 00:12:31, Size: 5252
mid05tor12cn16: [97] Generated: 12/05/2017 00:27:42, Size: 5328
mid05tor12cn16: [98] Generated: 12/05/2017 20:51:49, Size: 7296
mid05tor12cn16: [99] Generated: 12/06/2017 16:23:10, Size: 68740
mid05tor12cn16: [100] Generated: 12/06/2017 20:37:32, Size: 69144
mid05tor12cn16: [101] Generated: 12/07/2017 00:25:25, Size: 5108
mid05tor12cn16: [102] Generated: 12/07/2017 01:51:43, Size: 4832
mid05tor12cn15: [57] Generated: 11/30/2017 21:12:43, Size: 91868
mid05tor12cn15: [58] Generated: 12/01/2017 02:29:52, Size: 92792
mid05tor12cn15: [59] Generated: 12/01/2017 02:36:49, Size: 93000
mid05tor12cn15: [60] Generated: 12/01/2017 02:41:55, Size: 93340
mid05tor12cn15: [61] Generated: 12/01/2017 02:46:41, Size: 93652
mid05tor12cn15: [62] Generated: 12/01/2017 02:49:43, Size: 93852
mid05tor12cn15: [63] Generated: 12/01/2017 02:56:32, Size: 93988
mid05tor12cn15: [64] Generated: 12/01/2017 02:59:44, Size: 94244
mid05tor12cn15: [65] Generated: 12/01/2017 03:20:53, Size: 94520
mid05tor12cn15: [66] Generated: 12/04/2017 11:24:12, Size: 86152
mid05tor12cn15: [67] Generated: 12/05/2017 00:14:09, Size: 11768
mid05tor12cn15: [68] Generated: 12/05/2017 00:29:20, Size: 11840
mid05tor12cn15: [69] Generated: 12/05/2017 13:52:19, Size: 5040
mid05tor12cn15: [70] Generated: 12/05/2017 20:53:26, Size: 6688
mid05tor12cn02: [46] Generated: 11/30/2017 21:13:53, Size: 67604
mid05tor12cn02: [47] Generated: 11/30/2017 21:14:20, Size: 67636
mid05tor12cn02: [48] Generated: 11/30/2017 21:14:43, Size: 67668
mid05tor12cn02: [49] Generated: 12/01/2017 01:08:32, Size: 67552
mid05tor12cn02: [50] Generated: 12/01/2017 02:31:29, Size: 67608
mid05tor12cn02: [51] Generated: 12/01/2017 02:38:26, Size: 67564
mid05tor12cn02: [52] Generated: 12/01/2017 02:43:32, Size: 67560
mid05tor12cn02: [53] Generated: 12/01/2017 02:45:54, Size: 67608
mid05tor12cn02: [54] Generated: 12/01/2017 02:48:18, Size: 67576
mid05tor12cn02: [55] Generated: 12/01/2017 02:51:20, Size: 67600
mid05tor12cn02: [56] Generated: 12/01/2017 02:58:09, Size: 67620
mid05tor12cn02: [57] Generated: 12/01/2017 03:01:21, Size: 67552
mid05tor12cn02: [58] Generated: 12/01/2017 03:22:30, Size: 67600
mid05tor12cn02: [59] Generated: 12/07/2017 01:15:08, Size: 6524
mid05tor12cn02: [60] Generated: 12/07/2017 01:26:14, Size: 7424
mid05tor12cn02: [61] Generated: 12/07/2017 01:26:55, Size: 7932
mid05tor12cn02: [62] Generated: 12/07/2017 01:27:00, Size: 8816
mid05tor12cn02: [63] Generated: 12/07/2017 01:27:05, Size: 8988
mid05tor12cn02: [64] Generated: 12/07/2017 01:28:15, Size: 9364
mid05tor12cn02: [65] Generated: 12/07/2017 01:28:37, Size: 10128
mid05tor12cn02: [66] Generated: 12/07/2017 01:30:25, Size: 10824
mid05tor12cn02: [67] Generated: 12/07/2017 01:30:29, Size: 10904
mid05tor12cn02: [68] Generated: 12/07/2017 01:36:26, Size: 4944
mid05tor12cn02: [69] Generated: 12/07/2017 01:37:04, Size: 11620
mid05tor12cn02: [70] Generated: 12/07/2017 01:39:38, Size: 12376
mid05tor12cn02: [71] Generated: 12/07/2017 01:52:28, Size: 4952
mid05tor12cn02: [72] Generated: 12/07/2017 01:52:34, Size: 4948
mid05tor12cn02: [73] Generated: 12/07/2017 01:54:50, Size: 5156
```

It might be we have some issues on the BMC where it's not really clearing it?

**Comment by xuweibj** [comment]
Hi @whowutwut , I tried the newest code, it's ok. With Mark's fix #4470.
```
[root@briggs01 tmp]# rflash mid05tor12cn13 -a 3a7f1407
Thu Dec  7 22:17:01 2017 OpenBMC: [openbmc_debug]
Attempting to activate ID=3a7f1407, please wait...
Thu Dec  7 22:17:01 2017 mid05tor12cn13: [openbmc_debug] curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://172.11.139.13/login
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] login_response 200 OK
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X PUT -H "Content-Type: application/json" -d '{"data":"xyz.openbmc_project.Software.Activation.RequestedActivations.Active"}' https://172.11.139.13/xyz/openbmc_project/software/3a7f1407/attr/RequestedActivation
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] rflash_update_activate_response 200 OK
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.13/xyz/openbmc_project/software/3a7f1407
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] rflash_update_check_state_response 200 OK
Thu Dec  7 22:17:02 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X PUT -H "Content-Type: application/json" -d '{"data":false}' https://172.11.139.13/xyz/openbmc_project/software/3a7f1407/attr/Priority
Thu Dec  7 22:17:05 2017 mid05tor12cn13: [openbmc_debug] rflash_set_priority_response 200 OK
Thu Dec  7 22:17:05 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.13/xyz/openbmc_project/software/3a7f1407
Thu Dec  7 22:17:06 2017 mid05tor12cn13: [openbmc_debug] rflash_update_check_state_response 200 OK
mid05tor12cn13: Firmware activation successful.
-------------------------------------------------------
Firmware update complete: Total=1 Success=1 Failed=0
-------------------------------------------------------
[root@briggs01 tmp]# rpower mid05tor12cn13 bmcreboot
Thu Dec  7 22:17:18 2017 OpenBMC: [openbmc_debug]
Thu Dec  7 22:17:18 2017 mid05tor12cn13: [openbmc_debug] curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://172.11.139.13/login
Thu Dec  7 22:17:19 2017 mid05tor12cn13: [openbmc_debug] login_response 200 OK
Thu Dec  7 22:17:19 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.13/xyz/openbmc_project/software/enumerate
Thu Dec  7 22:17:19 2017 mid05tor12cn13: [openbmc_debug] rinv_firm_response 200 OK
mid05tor12cn13: Firmware will be flashed on reboot, deleting all BMC diagnostics...
Thu Dec  7 22:17:19 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X POST -H "Content-Type: application/json" -d '{"data":[]}' https://172.11.139.13/xyz/openbmc_project/dump/action/DeleteAll
Thu Dec  7 22:17:20 2017 mid05tor12cn13: [openbmc_debug] rspconfig_dump_clear_response 200 OK
Thu Dec  7 22:17:20 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X PUT -H "Content-Type: application/json" -d '{"data":"xyz.openbmc_project.State.BMC.Transition.Reboot"}' https://172.11.139.13/xyz/openbmc_project/state/bmc0/attr/RequestedBMCTransition
Thu Dec  7 22:17:23 2017 mid05tor12cn13: [openbmc_debug] rpower_reset_response 200 OK
mid05tor12cn13: BMC reboot
[root@briggs01 tmp]# rspconfig mid05tor12cn13 dump -l
Thu Dec  7 22:20:57 2017 OpenBMC: [openbmc_debug]
Thu Dec  7 22:20:58 2017 mid05tor12cn13: [openbmc_debug] curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://172.11.139.13/login
Thu Dec  7 22:20:59 2017 mid05tor12cn13: [openbmc_debug] login_response 200 OK
Thu Dec  7 22:20:59 2017 mid05tor12cn13: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.13/xyz/openbmc_project/dump/enumerate
Thu Dec  7 22:21:00 2017 mid05tor12cn13: [openbmc_debug] rspconfig_dump_list_response 200 OK
mid05tor12cn13: No attributes returned from the BMC.
```

**Comment by whowutwut** [comment]
Thanks so much for checking that! @xuweibj
